jose/jwk.py

```diff
@@ -245,7 +259,191 @@ def verify(self, msg, sig):
+        (1, 3, 132, 0, 35): 'secp521r1'
+    }
+
+    prepared_key = None
+    curve = None
+
+    def __init__(self, key, algorithm):
+        if algorithm not in self.valid_hash_algs:
+            raise JWKError('hash_alg: %s is not a valid hash '
+                           'algorithm', algorithm)
+        self.curve = self.curve_map.get(algorithm)
+        sha_map = {
+            'ES256': 'sha256',
+            'ES384': 'sha384',
+            'ES512': 'sha512',
+        }
+
+        if isinstance(key, dict):
+            self.prepared_key = self._process_jwk(key, sha_map[algorithm])
+            return
+
+        if isinstance(key, six.string_types):
+            if isinstance(key, six.text_type):
+                key = key.encode('utf-8')
+
+            # be a bit smart about what you're doing.
+            # keys must be in raw form, not ASN1, so convert if needed.
+
+            # The private key provided for testing is a base64 ASN1 that has a
+            # PEM wrapper. This may take a bit of guesswork...
+            der = self.pem_to_der(key)
+            # The key dictates the curve, this emulates the ecdsa lib
+            (self.curve, raw_key, raw_pub) = self.asn_to_raw(der, self.curve)
+            self.prepared_key = pyelliptic.ECC(
+                curve=self.curve,
+                privkey=raw_key,
+                pubkey=raw_pub,
+                hasher=sha_map[algorithm])
+            return
+        raise JWKError('Unable to parse an ECKey from key: %s' % key)
+
+    def repad(self, st):
+        """Add base64 padding back to the end of a stripped character
+        sequence
+
+        """
+        pad = '====' if isinstance(st, six.text_type) else b'===='
+        return st + pad[len(st) % 4:]
+
+    def pem_to_der(self, pem):
+        lines = pem.strip().split(b"\n")
+        return b''.join([line.strip() for line in lines if b'---' not in line])
+
+    def bitstring_to_str(self, bitstring):
+        """Convert an ASN1 BitString to a character array."""
+        if isinstance(bitstring, univ.OctetString):
+            return bitstring.asOctets()
+        if isinstance(bitstring, univ.BitString):
+            # Convert using a 2.7 safe method.
+            lh = hex(int(
+                ''.join(map(str, bitstring)), base=2))[2:].replace('L', '')
+            return unhexlify('0' * (len(lh) % 2) + lh)
+
+    def asn_to_raw(self, candidate, curve):
+        """Extract the ASN1 information and return the curve and key pairs."""
+        decoded = base64.urlsafe_b64decode(self.repad(candidate))
+
+        # if it's already raw... (Most likely a public key)
+        if len(decoded) == 64:
+            return curve, None, "\04" + decoded
+        if decoded[0] == "\04":
+            return curve, None, decoded
+
+        try:
+            asn_set = decoder.decode(decoded)[0]
+        except:
+            raise JWKError("Invalid EC Key")
+        pri_key = self.bitstring_to_str(asn_set[1])
```

Comment:
This is only applicable for the first case; in the `univ.Sequence` case you end up having to shuffle this around.

I'd recommend initializing this to `None`, too, and moving this set into the first case block.